### PR TITLE
Fix compile with 'noresumabledigest' tag

### PR DESCRIPTION
**Pull request by noxiouz**

### CODE REVIEW

**Comment by RichardScothern** [comment]
😱 LGTM

**Comment by dmcgowan** [comment]
LGTM

File: registry/storage/blobwriter_nonresumable.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 )
 
 // resumeHashAt is a noop when resumable digest support is disabled.
```
Comment (dmcgowan) [inline registry/storage/blobwriter_nonresumable.go]: comment no longer matches

**Comment by noxiouz** [comment]
Do I understand properly that if `noresumabledigest` is set, `(*linkedBlobStore).Resume` should not be called?

**Comment by dmcgowan** [comment]
@noxiouz it does appear to be called. It seems that after this change, it compiles but does not work.

I am not able to upload and you can see `(*linkedBlobStore).Resume` is called
https://gist.github.com/dmcgowan/e29ee92bde2c3e4d9ccba99c6dd7ec53

**Comment by dmcgowan** [comment]
Despite this not working, still LGTM, the feature should be compilable even if it is not working. That is a separate issue. @RichardScothern wdyt?

**Comment by noxiouz** [comment]
>  It seems that after this change, it compiles but does not work.

I'm sure that it worked in 2.3.1.
I've migrated my driver implementation for 2.4 `Storage` interface and met this issue. 
So if `Resume` is not expected to be called with `noresumabledigest` tag I will try to fix it

**Comment by RichardScothern** [comment]
I believe the problem was `storeHashState` returning an error from the non-resumable digest blob writer and returning an error.  This causes the request to fail and stops the data being flushed.

**Comment by noxiouz** [comment]
thanks!

**Pull request closed without merging**